### PR TITLE
Add logic for additional wise currencies

### PR DESCRIPTION
**Pull request by richardliang**
Closed in favor of https://github.com/zkp2p/zk-p2p/pull/376

### CODE REVIEW

File: client/src/components/Notary/NotarizationTable.tsx [hunk 1]
```diff
@@ -154,15 +154,15 @@ export const NotarizationTable: React.FC<NotarizationTableProps> = ({
...
+            metadata_copy: `${selectedRequest.subject} on ${selectedRequest.date}`,
             metadata_type_copy: 'history transaction receipt',
             transaction_type_copy: 'depositor registration'
             };
   
         case NotaryVerificationCircuit.TRANSFER:
           return {
             detected_copy: 'The following transaction was detected from your Wise account history',
-            metadata_copy: `€${selectedRequest.metadata} EUR on ${selectedRequest.date}`,
+            metadata_copy: `${selectedRequest.subject} on ${selectedRequest.date}`, // TODO: update this
```
Comment (asoong) [inline client/src/components/Notary/NotarizationTable.tsx:165]: Move parsing for `on` or `at` into `parseTimestamp`

File: client/src/helpers/types/paymentPlatform.ts [hunk 2]
```diff
@@ -34,33 +34,38 @@ export type PaymentPlatformType = typeof PaymentPlatform[keyof typeof PaymentPla
...
+  platformCurrencyIcons: string[];
 }
 
 export const paymentPlatformInfo: Record<PaymentPlatformType, PaymentPlatformData> = {
   [PaymentPlatform.VENMO]: {
     platformId: PaymentPlatform.VENMO,
     platformName: 'Venmo',
-    platformCurrency: 'USD',
-    flagSvg: americaFlagSvg
+    platformCurrencies: ['USD'],
```
Comment (asoong) [inline client/src/helpers/types/paymentPlatform.ts:46]: Make these a custom type as well

File: client/src/components/Liquidity/DepositsTable.tsx [hunk 3]
```diff
@@ -75,21 +77,30 @@ export const DepositsTable: React.FC = () => {
 
   useEffect(() => {
     let depositStoreToDisplay: DepositWithAvailableLiquidity[] = [];
+    let conversionCurrenciesToDisplay: string[] = [];
     switch (paymentPlatform) {
       case PaymentPlatform.VENMO:
         depositStoreToDisplay = venmoDepositStore ?? [];
+        conversionCurrenciesToDisplay = depositStoreToDisplay.map(() => 'USD')
```
Comment (asoong) [inline client/src/components/Liquidity/DepositsTable.tsx:84]: Use the new currency type here

File: client/src/components/Deposit/Input.tsx [hunk 4]
```diff
@@ -17,6 +18,8 @@ interface InputProps {
   inputLabel?: string;
   readOnly?: boolean;
   accessoryLabel?: string;
+  hasSelector?: boolean;
+  selectorDisabled?: boolean;
```
Comment (asoong) [inline client/src/components/Deposit/Input.tsx:22]: Double check if you need this

File: client/src/components/Deposit/DepositTable.tsx [hunk 5]
```diff
@@ -212,24 +214,35 @@ export const PositionTable: React.FC<PositionTableProps> = ({
 
   useEffect(() => {
     let depositsToDisplay: DepositWithAvailableLiquidity[] | null = [];
+    let conversionCurrenciesToDisplay: string[] = [];
     if (paymentPlatform) {
       switch (paymentPlatform) {
         case PaymentPlatform.VENMO:
-          depositsToDisplay = venmoDeposits;
+          depositsToDisplay = venmoDeposits ?? [];
+          conversionCurrenciesToDisplay = depositsToDisplay.map(() => 'USD')
```
Comment (asoong) [inline client/src/components/Deposit/DepositTable.tsx:222]: Use new currency type

**Review by asoong — APPROVED** [review]
Minor changes requested, looks good

File: client/src/contexts/wise/Liquidity/LiquidityProvider.tsx [hunk 6]
```diff
@@ -153,6 +157,7 @@ const LiquidityProvider = ({ children }: ProvidersProps) => {
         outstandingIntentAmount: depositData.outstandingIntentAmount,
         conversionRate: depositData.conversionRate,
         intentHashes: depositData.intentHashes,
+        receiveCurrencyId: depositData.receiveCurrencyId,
```
[the remark this answers is not in the document]
Comment (richardliang) [reply client/src/contexts/wise/Liquidity/LiquidityProvider.tsx:160]: https://github.com/zkp2p/zk-p2p/blob/2db9484ec87e2a8425e264a86a0ed57ab94d6c43/client/src/helpers/types/paymentPlatform.ts#L16

File: client/src/contexts/common/PlatformSettings/PlatformSettingsContext.ts [hunk 7]
```diff
@@ -6,6 +6,8 @@ import { PaymentPlatformType, PaymentPlatform, paymentPlatforms } from '@helpers
 interface PlatformSettingsValues {
   paymentPlatform?: PaymentPlatformType;
   setPaymentPlatform?: React.Dispatch<React.SetStateAction<PaymentPlatformType>>;
+  currencyIndex?: number;
```
[the remark this answers is not in the document]
Comment (richardliang) [reply client/src/contexts/common/PlatformSettings/PlatformSettingsContext.ts]: https://github.com/zkp2p/zk-p2p/blob/2db9484ec87e2a8425e264a86a0ed57ab94d6c43/client/src/helpers/types/paymentPlatform.ts#L53

File: client/src/contexts/common/PlatformSettings/PlatformSettingsContext.ts [hunk 8]
```diff
@@ -6,6 +6,8 @@ import { PaymentPlatformType, PaymentPlatform, paymentPlatforms } from '@helpers
...
+  setCurrencyIndex?: React.Dispatch<React.SetStateAction<number>>;
```
[the remark this answers is not in the document]
Comment (richardliang) [reply client/src/contexts/common/PlatformSettings/PlatformSettingsContext.ts]: https://github.com/zkp2p/zk-p2p/blob/2db9484ec87e2a8425e264a86a0ed57ab94d6c43/client/src/helpers/types/paymentPlatform.ts#L53

File: client/src/components/Notary/NotarizationTable.tsx (hunk 1 above)
Comment (richardliang) [reply client/src/components/Notary/NotarizationTable.tsx]: https://github.com/zkp2p/zk-p2p/pull/354/commits/4b5bd8e7ec570c56771cd0e8644fb31c10e33d8b

File: client/src/components/Deposit/Input.tsx (hunk 4 above)
Comment (richardliang) [reply client/src/components/Deposit/Input.tsx]: Removed

**Pull request closed without merging**